### ISSUE TITLE
No taskbar icon on Plasma Wayland

### ISSUE THREAD

**Issue by alsoGAMER** [trigg/Discover#288]
With this being written with GTK, you can check the implementation used [here](https://github.com/z411/trackma/blob/master/trackma/ui/gtk/application.py) to figure out why it isn't working properly.

**Comment by trigg** [comment]
https://github.com/z411/trackma/blob/7d69f70301eb97c3ad380fce6d53232f35dd9e9d/trackma/ui/gtk/statusicon.py#L77

Looks to me like it disables systray on wayland?

**Comment by alsoGAMER** [comment]
> https://github.com/z411/trackma/blob/7d69f70301eb97c3ad380fce6d53232f35dd9e9d/trackma/ui/gtk/statusicon.py#L77
> 
> Looks to me like it disables systray on wayland?

Yes, that's the code for the systray icon, but i'm talking about the taskbar one.

![image](https://user-images.githubusercontent.com/35269770/204025395-2b554137-56d9-4202-92a4-ebd6b69ef373.png)
The app with the wayland icon is the settings app for this overlay.

**Comment by trigg** [comment]
Ah, I misunderstood! Thanks for clarifying

**Comment by cprin21** [comment]
I'd like to add some context - there are no icons for the following areas - the taskbar icon as alsoGAMER indicated, the system tray where you can see I've hovered over with my cursor (completely functional just invisible), and lastly in the opened configuration window the icon is missing and replaced with the placeholder "⊘" character.

This is a problem with the version installed via pipx, the version installed via COPR on Fedora has its icons properly displayed while the version installed via Flatpak doesn't have a system tray icon at all which is inconvenient.

![Screenshot_20231006_110607](https://github.com/trigg/Discover/assets/107228497/9753e510-b5e1-42d4-8c4b-187205c3093f)

**Comment by cprin21** [comment]
Just as an update, copying and pasting the hicolor folder to ~/.local/share/icons/ where there already is a hicolor folder and accepting to merge causes the icons to show up in the app and in the system tray but like alsoGAMER opened the issue for, the taskbar icon is still the placeholder Wayland icon.

**Comment by trigg** [comment]
pipx continues to not copy icon assets to the outer world.

It's fine using pipx for testing but we need a different suggestion for primary run-from-github section of readme

**Comment by trigg** [comment]
Just to confirm the position that pipx takes on passing the `.desktop` and icon files.

https://github.com/pypa/pipx/issues/717
